api/progress.go

```diff
@@ -0,0 +1,51 @@
+package api
+
+import (
+	"TUM-Live/dao"
+	"TUM-Live/tools"
+	"github.com/gin-gonic/gin"
+	log "github.com/sirupsen/logrus"
+	"net/http"
+)
+
+func configProgressRouter(router *gin.Engine) {
+	router.POST("/api/progressReport", saveProgress)
+}
+
+type ProgressRequest struct {
```

Comment:
I'm not a good example, but exported fields, methods, functions, and structs should have a comment documenting what they do. (I marked them with a .)